client/src/components/HackathonForm.jsx

```diff
@@ -0,0 +1,163 @@
+                ) : (
+                  <></>
+                )}
+                {/* eslint-disable-next-line */}
+                {values.categories.map((category, index) => {
+                  return (
+                    <div>
+                      <Grid
+                        container
+                        direction="row"
+                        justify="flex-start"
+                        alignItems="baseline"
+                      >
+
+                        <TextField
+                          className={classes.categoryField}
+                          name={`categories[${index}].name`}
+                          value={category.name}
+                          onChange={handleChange}
+                          label="Name"
+                          helperText="Name of the Category"
+                          required
+                        />
+
+                        <TextField
+                          className={classes.companyField}
+                          name={`categories[${index}].companyName`}
+                          value={category.companyName}
+                          onChange={handleChange}
+                          label="Company Name"
+                          helperText="Name of the company sponsoring the category (if Applicable)"
+                        />
+
+                        <TextField
+                          className={classes.typeField}
+                          name={`categories[${index}].type`}
+                          value={category.type}
+                          onChange={handleChange}
+                          label="Type"
+                          helperText="Type of the category, either 'Track' or 'Superlative'"
+                          required
+                        />
+
+                        <IconButton
+                          className={classes.deleteButton}
+                          aria-label="delete"
+                          color="secondary"
+                          onClick={() => remove(index)}
+                          edge="start"
+                        >
+                          <DeleteIcon />
+                        </IconButton>
+                        
+                      </Grid>
+                    </div>
+                  )
+                })}
+
+                <Button
+                  className={classes.button}
+                  type="button"
+                  onClick={() => push({ name: '', companyName: '', type: '',})}
+                >
+                  {values.categories.length === 0 ? 'Add link' : 'Add another link'}
+                </Button>
+                  </>
+                )}
+              </FieldArray>
+
+              <Button
+                name="submitButton"
+                type="submit"
+                variant="contained"
+                color="primary"
+              >
+                Submit
+              </Button>
+
+            </Form>
+          )}
+        </Formik>
+    );
+}
```

Comment:
In general, I found the layout of this form to be a little weird- it would be best if all the fields for each category were in a single row, instead of the third input being forced to the new line.

Here's what I see:
![image](https://user-images.githubusercontent.com/55605862/134760699-5670886a-4591-4fa8-992b-547fdaff723a.png)